src/System.Runtime/tests/System/Text/StringBuilderTests.cs

```diff
@@ -732,6 +732,19 @@ public static void Append_CharArray_InvalidDesktop()
             AssertExtensions.Throws<ArgumentOutOfRangeException>("requiredLength", () => builder.Append(new char[] { 'a' }, 0, 1));
         }
 
+        [Theory]
+        [InlineData("Hello", new char[] { 'a' }, "Helloa")]
+        [InlineData("Hello", new char[] { 'b', 'c', 'd' }, "Hellobcd")]
+        [InlineData("", new char[] { 'e', 'f', 'g' }, "efg")]
+        [InlineData("Hello", new char[0], "Hello")]
+        public static void Append_CharSpan(string original, char[] value, string expected)
+        {
+            var builder = new StringBuilder(original);
+            builder.Append(new ReadOnlySpan<char>(value));
+            Assert.Equal(expected, builder.ToString());
+            Assert.Equal(expected, builder.ToString());
```

Comment:
What's the reason for the duplicated line?  Validating that ToString is idempotent?  I'd hope that'd be covered in some test elsewhere, in which case it shouldn't also be necessary in this test focused on Append(Span)